client/signup/steps/website-content/index.tsx

```diff
@@ -84,19 +84,17 @@ function WebsiteContentStep( {
 	const translatedPageTitles = useTranslatedPageTitles();
 
 	useEffect( () => {
-		if ( websiteContent.pages.length > 0 ) {
```

Comment:
I didn't test this but, Doesn't this clear the saved content every-time we land on this page? The "Saving" concept is completely removed here right?

I think a better quick fix is to save the current active siteId and clear the pages whenever there is a different site.
So we cache page details only one site per person (For now).